docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb

```diff
@@ -0,0 +1,402 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Quantum teleporation\n",
+    "\n",
+    "\n",
+    "Below we shall use the EPR states to study an interesting quantum phenomena - teleportation.\n",
+    "\n",
+    "Let us start by implementing the circuit shown below and discuss its implications later. We normally assume that the input qubits are initialised to the $\\ket{0}$ state unless specified otherwise. In this case however, the circuit diagram depicts the 0<sup>th</sup> qubit to be initialised in an arbitary quantum state, $\\ket{\\psi} = \\alpha\\ket{0} + \\beta\\ket{1} $, and the 1<sup>st</sup> and 2<sup>nd</sup> qubit to be in the state $\\ket{\\beta_{00}} = \\tfrac{1}{\\sqrt{2}} [\\ket{00} + \\ket{11}]$.\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![](images/teleportation.png)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cudaq \n",
+    "import numpy as np \n",
+    "np.random.seed(33)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We now replicate the circuit shown in the diagram above and step through each intermediate stage from psi0 to psi4\n",
+    "\n",
+    "kernel = cudaq.make_kernel()\n",
+    "qubits = kernel.qalloc(3) \n",
+    "\n",
+    "#psi0 - create an arbitary quantum state on qubit 0 \n",
+    "kernel.rx(parameter = np.random.uniform(0, 2*np.pi), target = qubits[0])\n",
+    "kernel.ry(parameter = np.random.uniform(0, 2*np.pi), target = qubits[0])\n",
+    "kernel.rz(parameter = np.random.uniform(0, 2*np.pi), target = qubits[0])\n",
+    "\n",
+    "# and a maximally entangled bell state on qubits 1 and 2 \n",
+    "kernel.h(qubits[1]) \n",
+    "kernel.cx(control=qubits[1], target=qubits[2])\n",
+    "\n",
+    "snapshot0 = cudaq.get_state(kernel) #store the state of the system for later use \n"
```

Comment:
nit: inline doc comments should be used sparingly (https://peps.python.org/pep-0008/#inline-comments). I don't believe any of these are necessary as in-line and can either be removed, or repurposed into a larger, more descriptive comment for the entire code cell.